applications/nrf5340_audio/src/bluetooth/le_audio_cis_headset.c

```diff
@@ -4,3 +4,281 @@
+					  struct bt_audio_ep *ep,
+					  enum bt_audio_pac_type type,
+					  struct bt_audio_capability *cap,
+					  struct bt_codec *codec)
+{
+	struct bt_audio_stream *stream = &streams;
+
+	if (!stream->conn) {
+		LOG_DBG("ASE Codec Config stream %p", (void *)stream);
+		return stream;
+	}
+
+	LOG_WRN("No streams available");
+	return NULL;
+}
+
+static int stream_cap_reconfig_cb(struct bt_audio_stream *stream,
+			struct bt_audio_capability *cap,
+			struct bt_codec *codec)
+{
+	LOG_DBG("ASE Codec Reconfig: stream %p cap %p", (void *)stream, (void *)cap);
+
+	/* We only support one QoS at the moment, reject changes */
+	return -ENOEXEC;
+}
+
+static int stream_cap_qos_cb(struct bt_audio_stream *stream, struct bt_codec_qos *qos)
+{
+	LOG_DBG("QoS: stream %p qos %p", (void *)stream, (void *)qos);
+
+	return 0;
+}
+
+static int stream_cap_enable_cb(struct bt_audio_stream *stream,
+		      struct bt_codec_data *meta,
+		      size_t meta_count)
+{
+	int ret;
+	struct event_t event;
+
+	LOG_DBG("Enable: stream %p meta_count %u", (void *)stream, meta_count);
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_STREAMING;
+
+	ret = ctrl_events_put(&event);
+	ERR_CHK(ret);
+	return 0;
+}
+
+static int stream_cap_start_cb(struct bt_audio_stream *stream)
+{
+	LOG_DBG("Start: stream %p", (void *)stream);
+	return 0;
+}
+
+static int stream_cap_metadata_cb(struct bt_audio_stream *stream,
+			struct bt_codec_data *meta,
+			size_t meta_count)
+{
+	LOG_DBG("Metadata: stream %p meta_count %u", (void *)stream, meta_count);
+
+	return 0;
+}
+
+static int stream_cap_disable_cb(struct bt_audio_stream *stream)
+{
+	LOG_DBG("Disable: stream %p", (void *)stream);
+
+	return 0;
+}
+
+static int stream_cap_stop_cb(struct bt_audio_stream *stream)
+{
+	int ret;
+	struct event_t event;
+
+	LOG_INF("Stop: stream %p", (void *)stream);
+
+	event.event_source = EVT_SRC_LE_AUDIO;
+	event.le_audio_activity.le_audio_evt_type = LE_AUDIO_EVT_NOT_STREAMING;
+
+	ret = ctrl_events_put(&event);
+	ERR_CHK(ret);
+
+	return 0;
+}
+
+static int stream_cap_release_cb(struct bt_audio_stream *stream)
+{
+	LOG_INF("Release: stream %p", (void *)stream);
+	return 0;
+}
+static struct bt_audio_capability_ops stream_cap_codec_ops = {
+	.config = stream_cap_config_cb,
+	.reconfig = stream_cap_reconfig_cb,
+	.qos = stream_cap_qos_cb,
+	.enable = stream_cap_enable_cb,
+	.start = stream_cap_start_cb,
+	.metadata = stream_cap_metadata_cb,
+	.disable = stream_cap_disable_cb,
+	.stop = stream_cap_stop_cb,
+	.release = stream_cap_release_cb,
+};
+
+static void stream_recv(struct bt_audio_stream *stream, const struct bt_iso_recv_info *info,
+			   struct net_buf *buf)
+{
+	static uint32_t recv_cnt;
+	bool bad_frame = false;
+
+	if (receive_cb == NULL) {
+		LOG_ERR("The RX callback has not been set");
+		return;
+	}
+
+	if (!(info->flags & BT_ISO_FLAGS_VALID)) {
+		bad_frame = true;
+	}
+	//TODO: change the payload size according to the ISO setting
```

Comment:
This is not possible to do now?

Reply:
Why can we not use buf->len in stead of 120?

Reply:
This is because now the gateway is in stereo mode, and we only want to send left channel which is half of full packet

Reply:
```C
net_buf_add_mem(buf, data, 120);
```
This is from cis_gateway, so it seems we are sending 120 bytes, or did you mean the Android phone?

Reply:
I was mean to Android. But check this code with Android and nRF5340 audio gateway again, seems it works with buf->len now. I might fixed something or Android fixed something with latest update (1 day ago)